src/vsr/replica.zig

```diff
@@ -3344,6 +3427,47 @@ pub fn Replica(
             return true;
         }
 
+        /// Returns the op that will be `op_checkpoint` after the next checkpoint.
+        ///
+        /// For a replica with journal_slot_count=8 and lsm_batch_multiple=2:
+        ///
+        ///   |01|23|45|67| (initial log fill)
+        ///   |89|01|23|45| (first wrap of log)
+        ///   |67|89|01|23| (second wrap of log)
+        ///
+        ///   checkpoint() call      0   1   2
+        ///   op_checkpoint          0   5  11
+        ///   op_checkpoint_next     5  11  17
+        ///   op_checkpoint_trigger  7  13  19
+        ///
+        fn op_checkpoint_next(self: *const Self) u64 {
+            assert(self.op_checkpoint <= self.commit_min);
+            assert(self.op_checkpoint <= self.op);
+            assert(self.op_checkpoint == 0 or
+                (self.op_checkpoint + 1) % config.lsm_batch_multiple == 0);
+
+            const op = if (self.op_checkpoint == 0)
+                // First wrap: op_checkpoint_next = 8-2-1 = 5
+                config.journal_slot_count - config.lsm_batch_multiple - 1
```

Comment:
Assuming `lsm_batch_multiple = 4` here, wouldn't this be `8 - 4 - 1 = 3`?

Reply:
I use `lsm_batch_multiple = 2` in these examples (it is noted in this function's comment). I know it isn't a realistic value, but it is useful for illustration.